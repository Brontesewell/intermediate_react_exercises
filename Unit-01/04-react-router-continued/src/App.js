import React, { Component } from 'react';
import './App.css';
import { Switch, Route, Redirect, Link} from "react-router-dom";
import ColorList from "./ColorList";
import NewColor from "./NewColor";
import Color from "./Color";

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      colors: [
        {
          name: "red",
          hex: "#FF0000",
        }, 
        {
          name: "green",
          hex: "#00FF00",
        },
        {
          name: "blue",
          hex: "#0000FF"
        }
      ]
    }
    this.handleAdd = this.handleAdd.bind(this);
  }

  handleAdd(newColor) {
    this.setState({colors: [newColor, ...this.state.colors]})
  }

  render() {
    const colorListComponent = () => (
      <ColorList colors={this.state.colors} />
    );

    const newColorComponent = props => (
      <NewColor addColor={this.handleAdd} {...props} />
    )

    const colorComponent = props => {
      const colorName = props.match.params.color;
      const color = this.state.colors.find(color => color.name === colorName);
      return color ? <Color {...props} color={color} /> : <Redirect to="/colors" />;
    };

    return (
      <Switch>
        <Route exact path="/colors" render={colorListComponent} />
        <Route path="/colors/new" render={newColorComponent} />
        <Route path="/colors/:color" render={colorComponent} />
        <Redirect to="/colors" />
      </Switch>
    );
  }
}

export default App;
