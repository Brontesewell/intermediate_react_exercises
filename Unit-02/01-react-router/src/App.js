import React from 'react';
import MainContent from './MainContent';
import {Link} from 'react-router-dom';
import './App.css';

const HeaderNav = () => (
  <nav className="navbar navbar-default">
    <div className="container-fluid">
      <div className="navbar-header">
        <Link className="navbar-brand header-link" to="/">
          <span role="img" aria-label="nail polish icon">ğŸ’…</span> Makeup Mega Market <span role="img" aria-label="lipstick icon">ğŸ’„</span>
        </Link>
      </div>
      <ul className="nav navbar-nav">
        <li><Link className="header-link" to="/brands/almay">Brands</Link></li>
      </ul>
      <form></form>
      <ul className="nav navbar-nav navbar-right">
        <li><Link className="header-link" to="/cart"><span role="img" aria-label="shopping cart link">ğŸ›’</span></Link></li>
      </ul>
    </div>
  </nav>
);

const App = () => (
  <div>
    <HeaderNav/>
    <MainContent/>
  </div>
);

export default App;
