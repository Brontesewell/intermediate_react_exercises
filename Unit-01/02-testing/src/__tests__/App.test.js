import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import App from '../App';

it('renders without crashing', () => {
  shallow(<App />);
});

it('matches its snapshot', () => {
  const wrapper = shallow(<App />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
