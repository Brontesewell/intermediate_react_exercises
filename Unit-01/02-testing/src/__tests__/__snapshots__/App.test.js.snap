exports[`test matches its snapshot 1`] = `
<div
  className="App">
  <div
    className="App-header">
    <h2>
      What\'s going on?
    </h2>
  </div>
  <TodoList />
</div>
`;
